src/libstore/build/derivation-goal.cc

```diff
@@ -627,6 +627,7 @@ void DerivationGoal::tryToBuild()
     if (buildMode != bmCheck && allValid) {
         debug("skipping build of derivation '%s', someone beat us to it", worker.store.printStorePath(drvPath));
         outputLocks.setDeletion(true);
+        outputLocks.unlock();
```

Comment:
It's possible this might be better placed in `done`, or perhaps `buildDone` could be used (I'm not too familiar with the code).